app/javascript/lib/Entities/Board/Province.js

```diff
@@ -0,0 +1,65 @@
+import Entity from '../Entity';
+
+export default class Province extends Entity {
+  constructor(isLand, representation) {
+    super(representation);
+
+    this.isLand = isLand;
+
+    this.ownership = null;
+    this.hasArmsFactory = false;
+    this.hasNavalFactory = false;
+
+    this.friendlyUnits = new Map();
+    this.hostileUnits = new Map();
+  }
+
+  get isOcean() {
+    return this.isLand === false;
+  }
+  set isOcean(bool) {
+    this.isLand = bool === false;
+  }
+
+  get hasFactory() {
+    return this.hasArmsFactory || this.hasNavalFactory;
+  }
+  get isDevoidOfFactory() {
+    return this.hasFactory === false;
+  }
+
+  get isOccupiedByHostiles() {
+    return this.hostileUnits.size > 0;
+  }
+  get isNotOccupiedByHostiles() {
+    return this.isOccupiedByHostiles === false;
+  }
+}
+
+export const translateProvinceModel = (representation, allProvinces, allUnits, gameBoard) => {
+  // temporary until migration is complete
+  const oldProvinceModel = allProvinces.get(representation);
+  const geographicModel = gameBoard.graph.get(representation);
+
+  const province = new Province(
+    geographicModel.isOcean === false || geographicModel.isOcean === undefined,
+    representation,
+  );
+
+  province.ownership = geographicModel.nation;
+  province.hasArmsFactory = oldProvinceModel.factory === 'armaments';
+  province.hasNavalFactory = oldProvinceModel.factory === 'shipyard';
+
+  for (const [occupyingNation] of allUnits) {
+    const { armies, fleets, friendly } = allUnits
+      .get(occupyingNation)
+      .get(representation);
+
+    if (armies > 0 || fleets > 0) {
+      const unitPool = friendly ? province.friendlyUnits : province.hostileUnits;
```

Comment:
Where is `unitPool` used?

Reply:
The line right below it I think, that way we no longer have to perform the same logic twice

The alternative is
```
if (friendly) {
  province.friendlyUnits.set(occupyingNation, armies + fleets);
} else {
  province.hostileUnits.set(occupyingNation, armies + fleets);
}
```

Reply:
Ah I didn't notice that setting on `unitPool` sets the property of `province`. Great!